### PR TITLE
Split VHeader into VHeaderDesktop and VHeaderMobile components

### LINKED ISSUE #1830: Disable component auto-injection for duplicated named header components
## Problem

We currently get a ton of errors during build:

```
 WARN  Two component files resolving to the same name VSearchBar:                                                                        19:36:21

 - /home/sara/projects/openverse-frontend/src/components/VHeaderOld/VSearchBar/VSearchBar.vue
 - /home/sara/projects/openverse-frontend/src/components/VHeader/VSearchBar/VSearchBar.vue


 WARN  Two component files resolving to the same name VSearchButton:                                                                     19:36:21

 - /home/sara/projects/openverse-frontend/src/components/VHeaderOld/VSearchBar/VSearchButton.vue
 - /home/sara/projects/openverse-frontend/src/components/VHeader/VSearchBar/VSearchButton.vue
```

## Description

Let's disable component auto-injection for these components or better yet just disable it altogether.

## Additional context
Marked high because it could theoretically cause a probably hard-to-debug issue if auto-injection was used by accident.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1830 by @sarayourfriend

## Description

This PR creates a two components for search page mobile/desktop headers: `VHeaderMobile` (until `lg`) and `VHeaderDesktop`. 

It also removes the remaining components that cause the warning about duplicate components during the Nuxt build (`VMobileMenuModal`, `VHeaderFilter`, `VHeaderMenu`).

This is not creating the actual mobile header or the desktop header according to mockups. The main goal is to remove duplicate and unnecessary components, and to set the basis for future desktop and mobile header implementations.

## Testing Instructions

No components break without the `new_header` flag set on. So, the CI passing should be enough.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
@obulat could you update the description to include what are intentional changes here, and what are temporary omissions or visual bugs that will be rectified after merge? It's looking really great based on what I _think is_ supposed to be happening but I have a few things I'm not sure about. I think I can give this an approving review after getting some more information.

**Comment by obulat** [comment]
> @obulat could you update the description to include what are intentional changes here, and what are temporary omissions or visual bugs that will be rectified after merge? It's looking really great based on what I _think is_ supposed to be happening but I have a few things I'm not sure about. I think I can give this an approving review after getting some more information.

I've actually split this PR into several others: VSearchTypeButton (#1860), VFilterButton (#1861) and the cleanup of the components used for the search type popup/modal (#1842). So, I'm converting this PR to draft and will update it after those PRs are done.

File: src/components/VHeader/VHeaderMobile.vue
Comment (dhruvkb) [inline src/components/VHeader/VHeaderMobile.vue:11]: There are some height inconsistencies in the mobile header.
Screenshot 2022-09-29 at 11 54 08 AM

**Review by dhruvkb — APPROVED** [review]
I noticed a small height inconsistency but approving since the goal of the PR (which seems to be about splitting the mobile and desktop headers) is accomplished here.

File: src/components/VHeader/VHeaderMobile.vue
Comment (obulat) [reply src/components/VHeader/VHeaderMobile.vue:11]: Thank you, @dhruvkb ! I did not try to fix the mobile header, as you say, because it's not in scope of this PR.

**Review by zackkrida — APPROVED** [review]
Much simpler, lgtm.

**Pull request merged**
